src/XCCDF/xccdf_session.c

```diff
@@ -119,6 +119,48 @@ static void _xccdf_session_free_oval_result_sources(struct xccdf_session *sessio
 static const char *oscap_productname = "cpe:/a:open-scap:oscap";
 static const char *oval_sysname = "http://oval.mitre.org/XMLSchema/oval-definitions-5";
 
+struct xccdf_session *xccdf_session_new_from_source(struct oscap_source *source, struct oscap_source *result_source)
+{
+	if (source == NULL)
+		return NULL;
+
+	struct xccdf_session *session = (struct xccdf_session *) oscap_calloc(1, sizeof(struct xccdf_session));
+
+	session->source = source;
+	session->xccdf.result_source = result_source;
```

Comment:
It looks like result_source can be NULL but it isn't documented in the function docs.